src/drivers/display/drm/lv_linux_drm.c

```diff
@@ -1053,6 +1147,198 @@ static void drm_flush(lv_display_t * disp, const lv_area_t * area, uint8_t * px_
+    EGLDisplay display = lv_opengles_egl_window_get_display(window);
+
+    drm_dev->kms_in_fence_fd = drm_dev->eglDupNativeFenceFDANDROID(display, drm_dev->gpu_fence);
+    LV_ASSERT(drm_dev->eglDestroySyncKHR(display, drm_dev->gpu_fence));
+    drm_dev->gpu_fence = NULL;
+    LV_ASSERT(drm_dev->kms_in_fence_fd != -1);
+
+    drm_buffer_t * fb;
+    uint32_t i = 0;
+    uint32_t flags = DRM_MODE_ATOMIC_NONBLOCK;
+    int64_t start_time, report_time, cur_time;
+    int ret;
+
+    /* Get the next bo to display */
+    struct gbm_bo * next_bo = gbm_surface_lock_front_buffer(drm_dev->surface);
+    if(!next_bo) {
+        LV_LOG_ERROR("Failed to lock frontbuffer");
+    }
+
+    fb = drm_fb_get_from_bo(next_bo);
+    if(!fb) {
+        LV_LOG_ERROR("Failed to get a new framebuffer BO");
+    }
+
+    if(drm_dev->kms_fence) {
+        EGLint status;
+
+        do {
+            status = drm_dev->eglClientWaitSyncKHR(display,
+                                                   drm_dev->kms_fence,
+                                                   0,
+                                                   EGL_FOREVER_KHR);
+        } while(status != EGL_CONDITION_SATISFIED_KHR);
+
+        LV_ASSERT(drm_dev->eglDestroySyncKHR(display, drm_dev->kms_fence));
+        drm_dev->kms_fence = NULL;
+    }
+
+    ret = drm_atomic_commit(drm_dev, fb->fb_id, flags);
+    if(ret) {
+        LV_LOG_ERROR("failed to commit: %s", strerror(errno));
+    }
+
+    /* release last buffer to render on again: */
+    if(drm_dev->bo)
+        gbm_surface_release_buffer(drm_dev->surface, drm_dev->bo);
+    drm_dev->bo = next_bo;
+
+}
+
+
+static drm_buffer_t * drm_fb_get_from_bo(struct gbm_bo * bo)
+{
+    int drm_fd = gbm_device_get_fd(gbm_bo_get_device(bo));
+    drm_buffer_t * fb = gbm_bo_get_user_data(bo);
+    uint32_t width, height, format, strides[4] = {0}, handles[4] = {0}, offsets[4] = {0}, flags = 0;
+    int ret = -1;
+
+    if(fb)
+        return fb;
+
+    fb = lv_malloc_zeroed(sizeof(drm_buffer_t));
+
+    width = gbm_bo_get_width(bo);
+    height = gbm_bo_get_height(bo);
+    format = gbm_bo_get_format(bo);
+    const int num_planes = gbm_bo_get_plane_count(bo);
+    for(int i = 0; i < num_planes; i++) {
+        handles[i] = gbm_bo_get_handle_for_plane(bo, i).u32;
+        strides[i] = gbm_bo_get_stride_for_plane(bo, i);
+        offsets[i] = gbm_bo_get_offset(bo, i);
+    }
+
+    memcpy(handles, (uint32_t [4]) {
+        gbm_bo_get_handle(bo).u32, 0, 0, 0
+    }, 16);
+    memcpy(strides, (uint32_t [4]) {
+        gbm_bo_get_stride(bo), 0, 0, 0
+    }, 16);
+    memset(offsets, 0, 16);
+    ret = drmModeAddFB2(drm_fd, width, height, format,
+                        handles, strides, offsets, &fb->fb_id, 0);
+    fb->fb_handle = fb->fb_id;
+
+    if(ret) {
+        LV_LOG_ERROR("failed to create fb: %s", strerror(errno));
+        lv_free(fb);
+        return NULL;
+    }
+
+    gbm_bo_set_user_data(bo, fb, drm_fb_destroy_callback);
+
+    return fb;
+}
```

Comment:
Nit: These sections seems not trivial and deserves at least a short comment on what each step does.